extension/ringmembershipextension/ringmembershipextension.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ringmembershipextension
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/facette/natsort"
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/extension"
+)
+
+type ringMembershipExtension struct {
+	sync.RWMutex
+	config Config
+	logger *zap.Logger
+	ipList []string
+	quit   chan int
+}
+
+var _ (extension.SupportExtension) = (*ringMembershipExtension)(nil)
+var _ (extension.ServiceExtension) = (*ringMembershipExtension)(nil)
+
+func (rm *ringMembershipExtension) serveState(w http.ResponseWriter, r *http.Request) {
+	m, err := json.Marshal(rm.ipList)
+	if err != nil {
+		w.WriteHeader(404)
+		return
+	}
+	w.Write(m)
+	return
+}
+
+// Starts the ring membership extension
+func (rm *ringMembershipExtension) Start(host component.Host) error {
+	rm.logger.Info("Starting ring membership extension")
+
+	rm.quit = make(chan int, 1)
+
+	go rm.Watch()
+
+	// Also start serving state at "GET /state"
+	go func() {
+		http.HandleFunc("/state", rm.serveState)
+		http.ListenAndServe("127.0.0.1:13000", nil)
+	}()
+	return nil
+}
+
+func (rm *ringMembershipExtension) Watch() error {
+	t := time.NewTicker(time.Second)
```

Comment:
What's the rationale for choosing 1 second interval and hard-coding it? Was any experimentation or analysis done to justify the choice?

Reply:
I am not sure how to converge on a good value for this, what experimentation/analysis do you recommend?